src/core/number_types.cpp

```diff
@@ -0,0 +1,278 @@
+#include "number_types.hpp"
+#include <algorithm>
+#include <type_traits>
+#include <cstdlib>
+
+std::map<CBType, NumberType> getCBTypeToNumberTypeMap() {
+  // clang-format off
+  return {
+    {Int, NumberType::Int64},
+    {Int2, NumberType::Int64},
+    {Int3, NumberType::Int32},
+    {Int4, NumberType::Int32},
+    {Int8, NumberType::Int16},
+    {Int16, NumberType::Int8},
+    {Float, NumberType::Float64},
+    {Float2, NumberType::Float64},
+    {Float3, NumberType::Float32},
+    {Float4, NumberType::Float32},
+    {Color, NumberType::UInt8},
```

Comment:
nit pick: let's keep the same order everywhere for easier maintenance.

* see number_types.hpp lines 10 to 16
* see number_types.cpp lines 49 to 55